usermods/Ble2Json_v2/usermod_v2_ble2json.h

```diff
@@ -0,0 +1,381 @@
+ * Multiple v2 usermods can be added to one compilation easily.
+ * 
+ * Creating a usermod:
+ * This file serves as an example. If you want to create a usermod, it is recommended to use usermod_v2_empty.h from the usermods folder as a template.
+ * Please remember to rename the class and file to a descriptive name.
+ * You may also use multiple .h and .cpp files.
+ * 
+ * Using a usermod:
+ * 1. Copy the usermod into the sketch folder (same folder as wled00.ino)
+ * 2. Register the usermod by adding #include "usermod_filename.h" in the top and registerUsermod(new MyUsermodClass()) in the bottom of usermods_list.cpp
+ */
+
+//class name. Use something descriptive and leave the ": public Usermod" part :)
+class Ble2JsonUsermod : public Usermod, Ble2JsonCallbacks {
+  private:
+    //Private class members. You can declare variables and functions only accessible to your usermod here
+    unsigned long lastTime = 0;
+
+    bool bleOnFlag = false;
+    uint32_t blePairingPin = PASSKEY;
+
+    bool bleInitted = false;
+    bool configDirty = false;
+
+    void checkBleInit(bool fromSetup) {
+      Serial.println("checkBleInit");
+
+      if (fromSetup && bleOnFlag && !bleInitted) {
+        Serial.println("bleInitting");
+        WLED::instance().disableWiFi();
+        bleInit(new MyCallbackHandler(this));
+        bleInitted = true;
+      }
+      else if (!fromSetup && bleOnFlag && !bleInitted) {
+        WLED::instance().reset();
+      }
+      else if (!bleOnFlag && bleInitted) {
+        Serial.println("ble on reset");
+        // TODO bleTearDown? and enableWiFi
+        WLED::instance().reset();
+      }
+    }
+
+  public:
+    //Functions called by WLED
+
+    /*
+     * setup() is called once at boot. WiFi is not yet connected at this point.
+     * You can use it to initialize variables, sensors or similar.
+     */
+    void setup() {
+      Serial.printf_P(PSTR("PIN: %u\n"),blePairingPin);
+
+      checkBleInit(true);
+    }
+
+
+    /*
+     * connected() is called every time the WiFi is (re)connected
+     * Use it to initialize network interfaces
+     */
+    void connected() {
+      // Serial.println("Connected to WiFi!");
+    }
+
+
+    /*
+     * loop() is called continuously. Here you can check for events, read sensors, etc.
+     * 
+     * Tips:
+     * 1. You can use "if (WLED_CONNECTED)" to check for a successful network connection.
+     *    Additionally, "if (WLED_MQTT_CONNECTED)" is available to check for a connection to an MQTT broker.
+     * 
+     * 2. Try to avoid using the delay() function. NEVER use delays longer than 10 milliseconds.
+     *    Instead, use a timer check as shown here.
+     */
+    void loop() {
+      if (millis() - lastTime > 3000) {
+        // Serial.println("I'm alive!");
+        lastTime = millis();
+
+        if (configDirty) {
+          serializeConfig();
```

Comment:
Where do you get actual configuration from?

Reply:
the changes happens in response to seeing a `"bleToggle":true` in the state... it is a transient state flag that toggles between ble and wifi mode.